client/main/variablesview.js

```diff
@@ -0,0 +1,784 @@
+
+        this.$el.html(`
+            <div class="jmv-variables-searchbox">
+                <div class="image"></div>
+                <input class="search" placeholder="Search variables"></input>
+            </div>
+            <div class="jmv-variables-container">
+                <div class="jmv-variables-body">
+
+                </div>
+            </div>`);
+
+        this.statusbar = new Statusbar();
+        this.$el.append(this.statusbar.$el);
+        this.statusbar.addInfoLabel('editStatus', { dock: 'left', value: 'Ready' });
+        this.statusbar.addActionButton('editFilters', { dock: 'left' });
+        this.statusbar.addActionButton('toggleFilterVisible', { dock: 'left' });
+        this.statusbar.addInfoLabel('activeFilters', { dock: 'left', label: 'Filters', value: 0 });
+        this.statusbar.addInfoLabel('columnCount', { dock: 'right', label: 'Variables', value: 0 });
+        this.statusbar.addInfoLabel('selectedCount', { dock: 'right', label: 'Selected', value: 0 });
+
+        this.$body      = this.$el.find('.jmv-variables-body');
+        this.$container = this.$el.find('.jmv-variables-container');
+
+        let $newVariable = $('<div class="add-new-variable"><span class="mif-plus"></span></div>');
+        this.$container.append(this._createCell($newVariable, 0, 1, 'new-variable', false, 4));
+
+        $newVariable.on('click', (event) => {
+            ContextMenu.showAppendVariableMenu($newVariable.offset().left + $newVariable.outerWidth(false) + 10, $newVariable.offset().top + $newVariable.outerHeight(false) - 10, 'right');
+        });
+
+        this.$search      = this.$el.find('.search');
+
+        this.$search.on('input', (event) => {
+            this._updateList();
+            if (this._topIndex !== -1)
+                this.selection.setSelection(0, this._topIndex, true);
+        });
+
+        this.$search.on('focus', () => {
+            keyboardJS.pause('variable-search');
+            this.$search.select();
+        });
+
+        this.$search.on('blur', () => {
+            keyboardJS.resume('variable-search');
+        });
+
+        this.$body.on('mouseleave', (event) => {
+            this.$body.find('.cell.hovering').removeClass('hovering');
+        });
+
+        this.controller = options.controller;
+        this.controller.registerView('variables', this);
+        this.selection = options.controller.selection;
+        this.selectionIncludesHidden = true;
+
+        this.selection.registerChangeEventHandler((oldSel, silent, ignoreTabStart) => {
+            if (this.rows.length === 0)
+                return;
+
+            this.$el.find('.selected .text').removeAttr('contenteditable');
+            this.$el.find('.cell.selected .text.name').removeAttr('placeholder');
+            this.$el.find('.cell.selected .text.description').removeAttr('placeholder');
+            this.$el.find('.selected').removeClass('selected');
+            let $current = this.$body.find('.select:checked');
```

Comment:
there's a lot of `.find()`s in this file ... here and further down ... can we move these queries to a model or controller, rather than the DOM?

Reply:
done